connector/core/core-service/src/main/java/org/oaebudt/edc/core/store/SqlParticipantGroupStore.java

```diff
@@ -0,0 +1,129 @@
+    public ParticipantGroup findById(String groupId) {
+        return transactionContext.execute(() -> {
+            try (var connection = getConnection()) {
+                Set<String> participantIds = queryExecutor.query(connection, false, this::mapParticipant, this.findParticipantsByGroupId(), groupId)
+                        .collect(Collectors.toSet());
+
+                return new ParticipantGroup(groupId, participantIds);
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+    @Override
+    public void save(String groupId, String... participantIds) {
+        transactionContext.execute(() -> {
+            try (var connection = getConnection()) {
+                queryExecutor.execute(connection, getUpsertParticipantGroupStatement(), groupId);
+
+                if(participantIds.length > 0) {
+                    queryExecutor.execute(connection, getUpsertMultipleParticipantStatement(groupId, participantIds));
+                }
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+    @Override
+    public Set<ParticipantGroup> getAllGroups() {
+
+        return transactionContext.execute(() -> {
+
+            try (var connection = getConnection()) {
+
+                Map<String, Set<String>> grouped =queryExecutor.query(connection, false, this::mapParticipantGroup, this.findAllParticipants())
+                        .collect(Collectors.groupingBy(
+                                arr -> arr[0], // key
+                                Collectors.mapping(
+                                        arr -> arr[1], // value
+                                        Collectors.filtering(
+                                                Objects::nonNull,
+                                                Collectors.toSet()
+                                        )
+                                )
+                        ));
+
+                Set<ParticipantGroup> result = new HashSet<>();
+                for (Map.Entry<String, Set<String>> entry : grouped.entrySet()) {
+                    result.add(new ParticipantGroup(entry.getKey(), entry.getValue()));
+                }
+                return result;
+            } catch (SQLException e) {
+                throw new EdcPersistenceException(e);
+            }
+        });
+    }
+
+
+    public String getUpsertParticipantGroupStatement() {
+        return "INSERT INTO %s (%s) VALUES (?) ON CONFLICT DO NOTHING;".formatted(PARTICIPANT_GROUP_TABLE, "id");
+    }
+
+    public String findParticipantsByGroupId() {
+        return "SELECT %s FROM %s WHERE %s = ?".formatted(PARTICIPANT_ID_COLUMN, PARTICIPANT_TABLE, GROUP_ID_COLUMN);
+    }
+
+    public String findAllParticipants() {
+        return "SELECT pg.id, participant_id " +
+                "FROM participant_group pg " +
+                "LEFT JOIN participant p ON " +
```

Comment:
I would suggest to use postgresql JSONB data type instead to create multiple tables and joins for no real need. that will make the storage more similar to the mongo one

Reply:
Changed to JsonB impl